pkg/compactor/blocks_cleaner.go

```diff
@@ -154,6 +175,10 @@ type cleanerJob struct {
 	timestamp int64
 }
 
+func (c *BlocksCleaner) SetRingLifecyclerID(ringLifecyclerID string) {
```

Comment:
Can we keep this method private? I see all other methods of `BlocksCleaner` are private